scripts/quests/ahtUrhgan/Delivering_the_Goods.lua

```diff
@@ -4,102 +4,121 @@
             },
-            ['Qutiba'] = {
+            ['Qutiba'] =
+            {
                 onTrigger = function(player, npc)
                     return quest:progressEvent(51)
                 end,
             },
-            ['Ulamaal'] = {
+            ['Ulamaal'] =
+            {
                 onTrigger = function(player, npc)
                     return quest:progressEvent(51)
                 end,
             },
 
-            onEventFinish = {
+            onEventFinish =
+            {
                 [39] = function(player, csid, option, npc)
                     quest:begin(player)
                 end,
             },
         },
     },
 
+    -- Section: Quest accepted
     {
         check = function(player, status, vars)
             return status == QUEST_ACCEPTED and vars.Prog == 0
         end,
 
-        [xi.zone.AHT_URHGAN_WHITEGATE] = {
-            ['Qutiba'] = {
+        [xi.zone.AHT_URHGAN_WHITEGATE] =
+        {
+            ['Qutiba'] =
+            {
                 onTrigger = function(player, npc)
                     return quest:progressEvent(40)
                 end,
             },
-            ['Ulamaal'] = {
+            ['Ulamaal'] =
+            {
                 onTrigger = function(player, npc)
                     return quest:progressEvent(40)
                 end,
             },
 
-            onEventFinish = {
+            onEventFinish =
+            {
                 [40] = function(player, csid, option, npc)
                     quest:setVar(player, 'Prog', 1)
                 end,
             },
         },
     },
 
+    -- Section: Complete quest
     {
         check = function(player, status, vars)
             return status == QUEST_ACCEPTED and vars.Prog == 1
         end,
 
-        [xi.zone.AHT_URHGAN_WHITEGATE] = {
-            ['Fochacha'] = {
+        [xi.zone.AHT_URHGAN_WHITEGATE] =
+        {
+            ['Fochacha'] =
+            {
                 onTrigger = function(player, npc)
                     return quest:progressEvent(41)
                 end,
             },
-            ['Qutiba'] = {
+            ['Qutiba'] =
+            {
                 onTrigger = function(player, npc)
                     return quest:progressEvent(52)
                 end,
             },
-            ['Ulamaal'] = {
+            ['Ulamaal'] =
+            {
                 onTrigger = function(player, npc)
                     return quest:progressEvent(53)
                 end,
             },
 
-            onEventFinish = {
+            onEventFinish =
+            {
                 [41] = function(player, csid, option, npc)
-                    player:setVar("Quest[6][62]Stage", getMidnight())
-                    player:needToZone(true)
-                    quest:complete(player)
+                    if quest:complete(player) then
+                        player:setVar("Quest[6][62]Stage", getMidnight())
+                        player:needToZone(true)
```

Comment:
from what i understood from interaction/container.lua, this would set a "quest-prefixed" local variable.
The actual check would be in a different quest though (in this case, Vanishing act)
How would I tell it to check for a previous quest variable with quest:getMustZone(player) in the new quest?